presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -90,14 +100,55 @@ private static boolean shouldRewriteTableLayout(TableScanNode source)
         @Override
         public Result apply(FilterNode filterNode, Captures captures, Context context)
         {
+            TableScanNode tableScan = captures.get(TABLE_SCAN);
+
             TableLayoutRewriter tableLayoutRewriter = new TableLayoutRewriter(metadata, context.getSession(), context.getSymbolAllocator(), context.getIdAllocator());
             PlanNode rewrittenTableScan = tableLayoutRewriter.planTableScan(captures.get(TABLE_SCAN), filterNode.getPredicate());
```

Comment:
`rewrittenTableScan` can be... a `FilterNode`, not a `TableScan`. Can you rename it to `rewritten` just as in `arePlansEquivalent()`?